ome_zarr/writer.py

```diff
@@ -432,8 +435,85 @@ def write_image(
         axes=axes,
         coordinate_transformations=coordinate_transformations,
         storage_options=storage_options,
+        **metadata,
+    )
+
+
+def write_multiscale_image_labels(
+    pyramid: List,
+    group: zarr.Group,
+    name: str,
+    chunks: Union[Tuple[Any, ...], int] = None,
+    fmt: Format = CurrentFormat(),
+    axes: Union[str, List[str], List[Dict[str, str]]] = None,
+    coordinate_transformations: List[List[Dict[str, Any]]] = None,
+    colors: List[JSONDict] = None,
+    properties: List[JSONDict] = None,
+    storage_options: Union[JSONDict, List[JSONDict]] = None,
+    **metadata: JSONDict,
+) -> None:
+    """
+    Write image labels in pyramidal format with multiscale and image-label metadata to disk.
+
+    Creates the label data in the sub-group "labels/{name}"
+
+    pyramid: List of np.ndarray
+      the image label data to save. Largest level first
+      All image arrays MUST be up to 5-dimensional with dimensions
+      ordered (t, c, z, y, x)
+    group: zarr.Group
+      the group within the zarr store to store the data in
+    name: str
+      the name of this labale data
+    chunks: int or tuple of ints,
+      size of the saved chunks to store the image
+    fmt: Format
+      The format of the ome_zarr data which should be used.
+      Defaults to the most current.
+    axes: str or list of str or list of dict
+      List of axes dicts, or names. Not needed for v0.1 or v0.2
+      or if 2D. Otherwise this must be provided
+    coordinate_transformations: 2Dlist of dict
+      For each path, we have a List of transformation Dicts.
+      Each list of dicts are added to each datasets in order
+      and must include a 'scale' transform.
+    colors: list of dict
+      Fixed colors for (a subset of) the label values.
+      Each dict specifies the color for one label and must contain the fields
+      "label-value" and "rgba".
+    properties: list of dict
+      Additional properties for (a subset of) the label values.
+      Each dict specifies additional properties for one label.
+      It must contain the field "label-value" and may contain arbitrary additional properties.
+    storage_options: dict or list of dict
+      Options to be passed on to the storage backend. A list would need to match
+      the number of datasets in a multiresolution pyramid. One can provide
+      different chunk size for each level of a pyramind using this option.
+    """
+    sub_group = group.require_group(f"labels/{name}")
+    write_multiscale(
+        pyramid,
+        sub_group,
+        chunks,
+        fmt,
+        axes,
+        coordinate_transformations,
+        storage_options,
+        name=name,
+        **metadata,
     )
-    group.attrs.update(metadata)
+
+    image_label_metadata = {}
```

Comment:
I assume this dictionary could also be assembled, combined with `**metadata` and passed to the `write_multiscale` call above?

Reply:
No that wouldn't work, since `**metadata` is written to `multiscales`. Whereas `image_label_metadata` is written to `image-label`, which is at the root of the json:
```yaml
{
"multiscales": [
  {
    ...  # this is where metadata goes
  }
],
"image-label": {
  ...  #  this is where image_label_metadata goes
}
}
```

Reply:
Ah I missed you changed the behavior of the API to have `**metadata` now appended  to the `multiscales` key level (as opposed to the top-level previously).

Is there another option for someone to add top-level keys e.g. `omero` and `image-label` as two specified examples?

Reply:
> Ah I missed you changed the behavior of the API to have **metadata now appended to the multiscales key level (as opposed to the top-level previously).

Yes, it's changed but I think the previous behavior was **very** unfortunate, since it would just write into top level without any connection to the multiscales metadata. So to add a property to multiscales you would need to do some very weird code gymnastics and it might even lead to incorrect metadata if you over-write a reserved key with something invalid.

Reply:
> Is there another option for someone to add top-level keys e.g. omero and image-label as two specified examples?

Not in the current function. But I think that for this use case just doing e.g. `group.attrs["omero"] = {...}` is acceptable.
Or if you want to be more explicit we could add explicit functions for this like `write_image_label_metadata` (could be added in this PR) and `write_omero_metadata` (this should be a different PR since it's unrelated). This would also mirror the current `write_multiscale_metadata` function.

Reply:
Following up, my initial expectation was that the current API could append multiscales metadata when passing a dictionary using a `multiscales` key. However running `write_image(data, self.group, axes="zyx", multiscales={"type": "gaussian"})` made me realize that the current implementation completely overwrites the `multiscales` metadata.

So :+1: for fixing this bug and defining an API with a set of tests allowing to store optional `multiscales` metadata esp. since several keys are defined in the spec. A few questions:
- what is the behavior if a core key (`axes`, `datasets`) is also passed as keywords?
- is there still room for an API allowing to defining other top-level keys or is the recommendation for consumers to invoke `group.attrs.update()`?